Fixes for managing content views chap

Levi-Leah opened the pull request:
Adds minor UI fixes to the Managing Content Views chapter. Removes procedure part of guides/common/modules/proc_resolving-package-dependencies.adoc and turns it into a concept module since according to BZ the entirety of "Content View Dependency Solving Algorithm" does not exist anymore.

Resolves the following BZ and Jiras:

[BZ#2123126](https://bugzilla.redhat.com/show_bug.cgi?id=2123126)
[BZ#2123128](https://bugzilla.redhat.com/show_bug.cgi?id=2123128)
[BZ#2123129](https://bugzilla.redhat.com/show_bug.cgi?id=2123129)
[BZ#2123130](https://bugzilla.redhat.com/show_bug.cgi?id=2123130)
[BZ#2123132](https://bugzilla.redhat.com/show_bug.cgi?id=2123132)
[BZ#2123134](https://bugzilla.redhat.com/show_bug.cgi?id=2123134)
[BZ#2123135](https://bugzilla.redhat.com/show_bug.cgi?id=2123135)
[BZ#2123142](https://bugzilla.redhat.com/show_bug.cgi?id=2123142)
[BZ#2123164](https://bugzilla.redhat.com/show_bug.cgi?id=2123164)


* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

adamlazik1 commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,14 +5,14 @@ Use this procedure to create a composite Content View.
 
 .Procedure
-. In the {ProjectWebUI}, navigate to *Content* > *Content Views* and click *Create New View*.
+. In the {ProjectWebUI}, navigate to *Content* > *Content Views* and click *Create content View*.
```
```suggestion
. In the {ProjectWebUI}, navigate to *Content* > *Content Views* and click *Create content view*.
```

adamlazik1 reviewed (commented):
Only found one small thing. Also, two additional notes:
* Note that one of the files changed in #1633, so you will need to rebase and resolve conflicts, then make changes as necessary.
* I would like to bring to your attention the typo in the PR title :slightly_smiling_face:

Levi-Leah commented:
My apologies for keeping this up for way too long. I was in training for part of last week. I'll add a few more fixes > rewrite the commit msg > rebase ASAP

Levi-Leah commented:
@maximiliankolb would you mind taking a look at the PR? I understand Content View Dependency Solving Algorithm described in  guides/common/modules/proc_resolving-package-dependencies.adoc does not exist anymore, could you verify if that's true for all versions?

adamlazik1 commented on guides/common/modules/proc_promoting-a-content-view.adoc:
```diff
@@ -23,8 +23,8 @@ You must assign both permissions to a user to allow them to promote Content View
 .Procedure
 . In the {ProjectWebUI}, navigate to *Content* > *Content Views* and select the Content View that you want to promote.
 . Click the *Versions* tab for the Content View.
```
```suggestion
```

adamlazik1 commented on guides/common/modules/proc_promoting-a-content-view.adoc:
```diff
@@ -23,8 +23,8 @@ You must assign both permissions to a user to allow them to promote Content View
 . Click the *Versions* tab for the Content View.
-. Select the version that you want to promote and in the *Actions* column, click *Promote*.
-. Select the environment where you want to promote the Content View, and click *Promote Version*.
+. Select the version that you want to promote and click the *More Options* icon > *Promote*.
```
```suggestion
. On the *Versions* tab, select the version that you want to promote and click on the vertical ellipsis on the right side of the screen.
. Click *Promote*.
```

Levi-Leah replied:
```suggestion
. Select the version that you want to promote, click the vertical ellipsis icon and click *Promote*.
```

adamlazik1 commented on guides/common/modules/proc_resolving-package-dependencies.adoc:
```diff
@@ -16,21 +16,3 @@ This might require some level of testing to determine what dependencies are requ
 If you add a filter that excludes some packages that are required and the Content View has dependency resolution enabled, {Project} ignores the rules you create in your filter in favor of resolving the package dependency.
 
 If you create a content filter for security purposes, to resolve a package dependency, {Project} can add packages that you might consider insecure.
-
-.Procedure
-
-To resolve package dependencies by default, complete the following steps:
-
-. In the {ProjectWebUI}, navigate to *Administer* > *Settings* and click the *Content* tab.
-. Locate the *Content View Dependency Solving Default*, and select *Yes*.
-
-You can also set the default level of dependency resolution that you want.
-You can select between adding packages to solve dependencies only if the required package does not exist, or to add the latest packages to resolve the dependency even if the package exists in the repository.
-
-To set the default level of dependency resolution, complete the following steps:
-
-. In the {ProjectWebUI}, navigate to *Administer* > *Settings* and click the *Content* tab.
```
I think there was an error maybe? You deleted the procedure and left the part which you also copied into the new file. Also, the two files have the same anchor so you will have to resolve that.

Levi-Leah replied:
> I think there was an error maybe? You deleted the procedure and left the part which you also copied into the new file. Also, the two files have the same anchor so you will have to resolve that.

I'll just delete the proc file if the changes are green-lighted, so no hustle with the IDs

adamlazik1 replied:
I see. My apologies.

maximiliankolb replied:
I am not following. Has this content completely been moved to `guides/common/modules/con_resolving-package-dependencies.adoc`?

maximiliankolb commented on guides/common/modules/con_resolving-package-dependencies.adoc:
```diff
@@ -0,0 +1,18 @@
+The package dependency resolution feature does not consider packages that are installed on your system independently of the Content View or solve dependencies across repositories.
+
+.Resolving Package Dependencies and Filters
+
```
```suggestion
```

maximiliankolb commented on guides/common/modules/con_resolving-package-dependencies.adoc:
```diff
@@ -0,0 +1,18 @@
+[id="Resolving_Package_Dependencies_{context}"]
+= Resolving Package Dependencies
+
+In {Project}, you can use the package dependency resolution feature to ensure that any dependencies that packages have within a Content View are added to the dependent repository as part of the Content View publication process.
```
```suggestion
You can resolve package dependencies to ensure that any dependencies that packages have within a Content View are added to the dependent repository as part of the Content View publication process.
```

maximiliankolb commented on guides/common/modules/con_resolving-package-dependencies.adoc:
```diff
@@ -0,0 +1,18 @@
+= Resolving Package Dependencies
+
+In {Project}, you can use the package dependency resolution feature to ensure that any dependencies that packages have within a Content View are added to the dependent repository as part of the Content View publication process.
+
```
```suggestion
```

If you keep this empty line, each sentence would stand of itself. IMHO a general rule is that a paragraph consists of at least two sentences.

maximiliankolb commented on guides/common/modules/con_resolving-package-dependencies.adoc:
```diff
@@ -0,0 +1,18 @@
+
+In {Project}, you can use the package dependency resolution feature to ensure that any dependencies that packages have within a Content View are added to the dependent repository as part of the Content View publication process.
+
+You can select to resolve package dependencies for any Content View that you want, or you can change the default setting to enable or disable resolving package dependencies for all new Content Views.
```
Is "enabled" or "disabled" the default setting? Either way, IMHO you should remove one aka. "or you can change the default setting to disable resolving package..." or "or you can change the default setting to enable resolving package...".

Levi-Leah replied:
> Is "enabled" or "disabled" the default setting?
disable seems to be the default. changing the wording appropriately

maximiliankolb commented on guides/common/modules/con_resolving-package-dependencies.adoc:
```diff
@@ -0,0 +1,18 @@
+This might require some level of testing to determine what dependencies are required.
+
+If you add a filter that excludes some packages that are required and the Content View has dependency resolution enabled, {Project} ignores the rules you create in your filter in favor of resolving the package dependency.
+
```
```suggestion
```

maximiliankolb commented on guides/common/modules/proc_promoting-a-content-view.adoc:
```diff
@@ -22,9 +22,8 @@ You must assign both permissions to a user to allow them to promote Content View
-. Click the *Versions* tab for the Content View.
-. Select the version that you want to promote and in the *Actions* column, click *Promote*.
-. Select the environment where you want to promote the Content View, and click *Promote Version*.
+. Select the version that you want to promote, click the vertical ellipsis icon and click *Promote*.
```
```suggestion
. Select the version that you want to promote, click the vertical ellipsis icon, and click *Promote*.
```

maximiliankolb commented on guides/common/modules/proc_promoting-a-content-view.adoc:
```diff
@@ -22,9 +22,8 @@ You must assign both permissions to a user to allow them to promote Content View
-. Select the version that you want to promote and in the *Actions* column, click *Promote*.
-. Select the environment where you want to promote the Content View, and click *Promote Version*.
+. Select the version that you want to promote, click the vertical ellipsis icon and click *Promote*.
+. Select the environment where you want to promote the Content View, and click *Promote*.
```
```suggestion
. Select the environment where you want to promote the Content View and click *Promote*.
```

Can you reword "where"? Sounds a bit off to me.

maximiliankolb reviewed (commented):
I made some suggestions. Please also rebase to master.

Levi-Leah commented:
@maximiliankolb apologies for the mess. PR should be up to date

maximiliankolb commented:
@Levi-Leah Please rebase to master and ensure your branch only contains the changes you want to make. It looks like there's at least one other commit about renaming images on your branch.

Levi-Leah commented:
@maximiliankolb I rebased but I don't see any 

> It looks like there's at least one other commit about renaming images on your branch.

@maximiliankolb I rebased but I don't see any commit abt renaming images in the log. I might be mistaken tho

maximiliankolb commented:
> @maximiliankolb I rebased but I don't see any
> 
> > It looks like there's at least one other commit about renaming images on your branch.
> 
> @maximiliankolb I rebased but I don't see any commit abt renaming images in the log. I might be mistaken tho

@Levi-Leah It might be caused by dadf7360502cf9bc73ce66ecbc8c868162d29020.

adamlazik1 commented:
@Levi-Leah looks like this is the last commit made. If you want to remove this commit do the following on your topic branch:
```
git reset --hard HEAD^
git push -f
```

adamlazik1 commented on guides/common/modules/proc_managing-ostree-content-with-content-views.adoc:
```diff
@@ -8,7 +8,7 @@ This process uses the same publication and promotion method as RPMs or Puppet mo
 
 .Procedure
-. In the {ProjectWebUI}, navigate to *Content* > *Content Views* and click *Create New View*.
+. In the {ProjectWebUI}, navigate to *Content* > *Content Views* and click *Create content View*.
```
```suggestion
. In the {ProjectWebUI}, navigate to *Content* > *Content Views* and click *Create content view*.
```

adamlazik1 commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,14 +5,14 @@ Use this procedure to create a composite Content View.
 . In the *Name* field, enter a name for the view.
 {ProjectName} automatically completes the *Label* field from the name you enter.
 . In the *Description* field, enter a description of the view.
 . Select the *Composite View?* checkbox to create a Composite Content View.
```
According to [BZ#2123134](https://bugzilla.redhat.com/show_bug.cgi?id=2123134), there is no **Composite View?** checkbox.

Levi-Leah replied:
> According to [BZ#2123134](https://bugzilla.redhat.com/show_bug.cgi?id=2123134), there is no **Composite View?** checkbox.

I double checked. the outlined steps work in the UI

Levi-Leah replied:
> According to [BZ#2123134](https://bugzilla.redhat.com/show_bug.cgi?id=2123134), there is no **Composite View?** checkbox.

good catch

adamlazik1 reviewed (commented):
I took a look at the bugzillas you mentioned in the PR message. Please take a look at [BZ#2123129](https://bugzilla.redhat.com/show_bug.cgi?id=2123129) and [BZ#2123134](https://bugzilla.redhat.com/show_bug.cgi?id=2123134) as the suggested changes don't seem to be included in this PR yet.

adamlazik1 reviewed (commented):
Made some minor suggestions. I would like to bring to your attention that [BZ#2123129](https://bugzilla.redhat.com/show_bug.cgi?id=2123129) still remains unaddressed, although I am not sure whether it is critical or even necessary. I only wanted to mention it since I see you included it in the PR message.

Levi-Leah commented:
@maximiliankolb I think this PR is ready to be merged

adamlazik1 commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,17 +5,20 @@ Use this procedure to create a composite Content View.
+. Optional: If you want to automatically update the Content View to the latest version, select the *Always update to latest version* checkbox.
+. Click *Add*, then click *Publish new version*.
+. Optional: In the *Description* field, enter a description of the Content View.
+. In the *Publish* pop-up window, rag the *Promote* slider, then select the lifecycle environment.
```
```suggestion
. In the *Publish* pop-up window, drag the *Promote* slider, then select the lifecycle environment.
```

adamlazik1 commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,17 +5,20 @@ Use this procedure to create a composite Content View.
+. Optional: If you want to automatically publish a new version of the Composite Content View when a Content View is republished, select the *Auto publish* checkbox.
+. Click *Create content view*.
+. In the *Content views* tab, select the Content Views that you want to add to the Composite Content View, and then click *Add content views*.
+. In the *Add content views* pop-up window, select the version of Content View.
```
```suggestion
. In the *Add content views* pop-up window, select the version of each Content View.
```

adamlazik1 reviewed (commented):
Apart from final suggestions (one typo), changes look good to me. Maximilian is on vacation AFAIK, but I can merge this for you. @Lennonka mind doing a double-check?

Levi-Leah commented:
> Maximilian is on vacation AFAIK, but I can merge this for you.
ty for letting me know, and sure, please merge it on my behalf

Lennonka commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,17 +5,20 @@ Use this procedure to create a composite Content View.
-In the *Description* field, enter a description and click *Save*.
-. Click *Promote* and select the lifecycle environments to promote the Composite Content View to, enter a description, and then click *Promote Version*.
+. Optional: In the *Description* field, enter a description of the view.
+. Under the *Type* tab, select *Composite content view*.
```
```suggestion
. On the *Type* tab, select *Composite content view*.
```

Lennonka commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,17 +5,20 @@ Use this procedure to create a composite Content View.
+. Under the *Type* tab, select *Composite content view*.
+. Optional: If you want to automatically publish a new version of the Composite Content View when a Content View is republished, select the *Auto publish* checkbox.
+. Click *Create content view*.
+. In the *Content views* tab, select the Content Views that you want to add to the Composite Content View, and then click *Add content views*.
```
```suggestion
. On the *Content views* tab, select the Content Views that you want to add to the Composite Content View, and then click *Add content views*.
```

Lennonka commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,17 +5,20 @@ Use this procedure to create a composite Content View.
+. Optional: If you want to automatically update the Content View to the latest version, select the *Always update to latest version* checkbox.
+. Click *Add*, then click *Publish new version*.
+. Optional: In the *Description* field, enter a description of the Content View.
+. In the *Publish* pop-up window, drag the *Promote* slider, then select the lifecycle environment.
```
```suggestion
. In the *Publish* pop-up window, set the *Promote* switch, then select the lifecycle environment.
```
Is "Promote" a slider (with more than two values) or a switch (true/false)?

adamlazik1 replied:
It is a switch with true/false. Good catch!

Lennonka commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,17 +5,20 @@ Use this procedure to create a composite Content View.
-. In the {ProjectWebUI}, navigate to *Content* > *Content Views* and click *Create New View*.
-. In the *Name* field, enter a name for the view.
+. In the {ProjectWebUI}, navigate to *Content* > *Content Views* and click *Create content view*.
+. In the *Create content view* pop-up window, enter a name for the view in the *Name* field.
```
I'm not sure if we should use "pop-up window", maybe I'd go with just "window".
(I haven't found any other usage of "pop-up window" in the docs apart from this PR.)

Levi-Leah replied:
```suggestion
. In the *Create content view* window, enter a name for the view in the *Name* field.
```

Lennonka commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,17 +5,20 @@ Use this procedure to create a composite Content View.
+. Optional: If you want to automatically publish a new version of the Composite Content View when a Content View is republished, select the *Auto publish* checkbox.
+. Click *Create content view*.
+. In the *Content views* tab, select the Content Views that you want to add to the Composite Content View, and then click *Add content views*.
+. In the *Add content views* pop-up window, select the version of each Content View.
```
```suggestion
. In the *Add content views* window, select the version of each Content View.
```

Lennonka reviewed (changes requested):
I've reviewed it by the IBM Style Guide. Just a couple of details, otherwise nice work!

adamlazik1 commented on guides/common/modules/proc_creating-a-composite-content-view.adoc:
```diff
@@ -5,17 +5,20 @@ Use this procedure to create a composite Content View.
+. Optional: If you want to automatically update the Content View to the latest version, select the *Always update to latest version* checkbox.
+. Click *Add*, then click *Publish new version*.
+. Optional: In the *Description* field, enter a description of the Content View.
+. In the *Publish* pop-up window, set the *Promote* switch, then select the lifecycle environment.
```
```suggestion
. In the *Publish* window, set the *Promote* switch, then select the lifecycle environment.
```
Since we decide to remove mentions to pop-up windows after all, please also remove this one. Once you've done that, this is good to go.

Levi-Leah commented:
> I've reviewed it by the IBM Style Guide. Just a couple of details, otherwise nice work!

@Lennonka thanks for the review. I believe I incorporated all of your suggestions

adamlazik1 reviewed (approved):
LGTM

adamlazik1 commented:
Merged and cherry-picked:
ac9636721..b6f481e15  3.1 -> 3.1
bc2d759b6..ff209f90c    3.3 -> 3.3
df5c25077..6864880db  3.4 -> 3.4
@Levi-Leah Could not cherry-pick to 3.2 because of merge conflicts. If you wish to have this change incorporated in 3.2 as well, please create a separate PR against 3.2 branch. I am not sure if it is necessary, though.

Levi-Leah commented:
> If you wish to have this change incorporated in 3.2 as well, please create a separate PR against 3.2 branch. I am not sure if it is necessary, though.

Thanks for merging this. I believe we can do without 3.2
